src/client/gui/gui_cmd.cpp

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (C) 2019 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "gui_cmd.h"
+#include "argparser.h"
+
+#include <multipass/cli/client_common.h>
+#include <multipass/cli/client_platform.h>
+#include <multipass/cli/format_utils.h>
+
+#include <QEventLoop>
+#include <QTimer>
+#include <QtConcurrent/QtConcurrent>
+
+namespace mp = multipass;
+namespace cmd = multipass::cmd;
+using RpcMethod = mp::Rpc::Stub;
+
+mp::ReturnCode cmd::GuiCmd::run(mp::ArgParser* parser)
+{
+    if (!QSystemTrayIcon::isSystemTrayAvailable())
+    {
+        cerr << "System tray not supported\n";
+        return ReturnCode::CommandFail;
+    }
+
+    QEventLoop loop;
```

Comment:
Is this necessary? There should be an event loop created by the QCoreApplication already that you can use.

Reply:
Sure, I can use the static `QCoreApplication::exec()` here instead of the explicit `QEventLoop`.  Accomplishes the same thing in effect, but is cleaner just using the static.

Reply:
Ok, pushed fix.